pkg/chains/formats/intotoite6/intotoite6.go

```diff
@@ -58,20 +59,59 @@ func (i *InTotoIte6) Wrap() bool {
 }
 
 func (i *InTotoIte6) CreatePayload(obj interface{}) (interface{}, error) {
-	var tr *v1beta1.TaskRun
 	switch v := obj.(type) {
 	case *v1beta1.TaskRun:
-		tr = v
+		return i.generateAttestationFromTaskRun(v)
+	case *v1beta1.PipelineRun:
+		return i.generateAttestationFromPipelinerun(v)
 	default:
 		return nil, fmt.Errorf("intoto does not support type: %s", v)
 	}
-	return i.generateAttestationFromTaskRun(tr)
+}
+
+// There most likely exists a better place to put this, adding as another method for now
+// Possibly a generic attestation interface?
+func (i *InTotoIte6) generateAttestationFromPipelinerun(pr *v1beta1.PipelineRun) (interface{}, error) {
+	// We don't need to pass in a client/context here, as we only want access to the original object
+	// Need a better way to differentiate between an abstracted original k8s object and getting the latest values
+	// - Possibly pass client and context in each method call instead of adding it to the struct?
+	pro := objects.NewPipelineRunObject(pr, nil, nil)
+
+	subjects := GetSubjectDigests(pro, i.logger)
+
+	// Adding a sample value to know the attestation is an example
+	invocation := slsa.ProvenanceInvocation{}
+	params := make(map[string]string)
+	params["message"] = "This is an example attestation for pipelineruns, we'll fill in the details later"
+	invocation.Parameters = params
+
+	att := intoto.ProvenanceStatement{
+		StatementHeader: intoto.StatementHeader{
+			Type:          intoto.StatementInTotoV01,
+			PredicateType: slsa.PredicateSLSAProvenance,
+			Subject:       subjects,
+		},
+		Predicate: slsa.ProvenancePredicate{
+			Builder: slsa.ProvenanceBuilder{
+				ID: i.builderID,
+			},
+			BuildType:  tektonID,
+			Invocation: invocation,
+			// BuildConfig: buildConfig(tr),
```

Comment:
This should probably be the run payload?

Reply:
We currently are creating an attestation that serves our needs while keeping in line with the provenance predicate. We want to keep in line with the schema, so i would like to discuss what we have so far in another thread. Perhaps the chains slack?